dashboard/app/views/layouts/_small_footer.html.haml

```diff
@@ -1,24 +1,32 @@
-.small-footer
-  - if view_options[:has_i18n]
-    = form_tag(locale_url, method: :post, id: 'localeForm', style: 'margin-bottom: 0px;') do
-      = hidden_field_tag :return_to, request.url
-      = select_tag :locale, options_for_select(options_for_locale_select, locale), onchange: 'this.form.submit();'
-  %small
-    = link_to I18n.t('footer.copyright'), '#', class: 'copyright-link'
-    &nbsp;&nbsp;|&nbsp;&nbsp;
-    %a{:href => "#", :class => "more-link"}
-      = I18n.t('footer.more')
-      %i{:class => "fa fa-caret-up"}
+= javascript_include_tag 'react'
 
-#copyright-flyout
-  %p!= I18n.t('footer.thank_you')
-  %p= I18n.t('footer.help_from_html')
-  %p!= I18n.t('footer.art_from_html', current_year: Time.now.year)
-  %p= I18n.t('footer.powered_by_aws')
-  %p!= I18n.t('footer.trademark', current_year: Time.now.year)
+#page-small-footer
 
-%ul#more-menu
-  %li= link_to I18n.t('footer.support'), 'https://support.code.org'
-  %li= link_to I18n.t('footer.translate'), 'https://code.org/translate'
-  %li= link_to I18n.t('footer.tos'), 'https://code.org/tos'
-  %li= link_to I18n.t('footer.privacy'), 'https://code.org/privacy'
+:javascript
```

Comment:
It's not necessary to make it a partial, unless you actually need to render
it inline. You can make it a js file and include it via script tag. See:

https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/views/levels/editors/_all.html.haml
https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/assets/javascripts/editor/blockly_editor.js.erb


On Thu, Aug 27, 2015 at 12:47 PM, davidsbailey <notifications@github.com>
wrote:

> In dashboard/app/views/layouts/_small_footer.html.haml
> <https://github.com/code-dot-org/code-dot-org/pull/3744#discussion_r38138101>
> :
>
> >
> > -%ul#more-menu
> > -  %li= link_to I18n.t('footer.support'), 'https://support.code.org'
> > -  %li= link_to I18n.t('footer.translate'), 'https://code.org/translate'
> > -  %li= link_to I18n.t('footer.tos'), 'https://code.org/tos'
> > -  %li= link_to I18n.t('footer.privacy'), 'https://code.org/privacy'
> > +:javascript
>
> Per our new effort, this code should go in a new javascript file. It
> should be possible to follow this example pretty closely:
> https://github.com/code-dot-org/code-dot-org/pull/3491/files
>
> If I understand how the rendering works, you would (for example) name the
> new file _small_footer_helper.js and then call #{render partial:
> 'layouts/small_footer_helper', formats: [:js]} from ruby.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/3744/files#r38138101>.
>